### PR TITLE
Add anycabled binary

### PR DESCRIPTION
**Pull request by sponomarev**
The motivation is simple - provide a way to deploy it with ~good old~ Capistrano.

I tried to minimize my changes as little as possible wrapping application into a separate daemon binary. Limitation - anycable cli args are not handled, I'll update the docs separately.

### CODE REVIEW

File: bin/anycabled [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env ruby
+
+require "anycable/cli"
+require "daemons"
+
+options = {
+  dir: "tmp/pids",
+  log_otput: false
```
Comment (sadovnik) [inline bin/anycabled]:
```suggestion
  log_output: false
```

**Comment by Envek** [comment]
I don't think it is necessary to have this functionality in the gem.

An application (or its infrastructure) should care about how to run and control daemons, whether it would be monit, systemd, upstart, docker-compose, kubernetes (all of these works best with foreground processes) or Capistrano calling a wrapper written with daemons gem or whatever else.

So, in my opinion, this wrapper script and gem dependency should go into an application.

**Comment by sponomarev** [comment]
@Envek cc @palkan The idea is to add all the Capistrano recipes on the next step. It's a pretty common thing to have them in a gem itself. I hope you don't need any examples. It's definitely a good thing to stimulate early adopters and new users. Frankly, I don't see any problems with that. If you worry about the gem size - my compensation is in #70 – 2Mb+ size reduction. If you worry about dependencay, we can make it optional and ask users to require it on their own if the need the binary (e.g. https://github.com/collectiveidea/delayed_job#running-jobs). But again, there is a high chance that you already have `daemons` in you dependencies.

File: bin/anycabled (hunk 1 above)
Comment (sponomarev) [reply bin/anycabled]: Appreciate!

File: bin/anycabled [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  dir: "tmp/pids",
+  log_output: false
+}
+
+# Preserve current directory. We need it inside the server.
+current_dir = Dir.pwd
+
+Daemons.run_proc("anycable", options) do
+  Dir.chdir current_dir
+  AnyCable::CLI.new.run
```
Comment (palkan) [inline bin/anycabled]: > anycable cli args are not handled

Looks like we can pass args to the process: https://github.com/thuehlinger/daemons/blob/master/lib/daemons.rb#L88

File: anycable.gemspec [hunk 3]
```diff
@@ -15,13 +15,14 @@ Gem::Specification.new do |spec|
...
 
-  spec.executables   = ["anycable"]
+  spec.executables   = %w[anycable anycabled]
   spec.files         = `git ls-files -z`.split("\x0").reject { |f| f.match(%r{^(test|spec|features)/}) }
   spec.require_paths = ["lib"]
 
   spec.required_ruby_version = '>= 2.4.0'
 
   spec.add_dependency "anyway_config", "~> 1.4.2"
+  spec.add_dependency "daemons", "~> 1.3"
```
Comment (palkan) [inline anycable.gemspec]: > we can make it optional and ask users to require it on their own

Let's do this)

We can add a user-friendly message to `anycabled` with the instructions on how to install the dependency.

See, for example:
- https://github.com/palkan/test-prof/blob/master/lib/test_prof/ruby_prof.rb#L204-L223
- https://github.com/palkan/test-prof/blob/master/lib/test_prof.rb#L50-L56
- https://github.com/palkan/test-prof/blob/master/lib/test_prof/utils.rb#L7-L16

Comment (sponomarev) [reply anycable.gemspec]: Done

File: bin/anycabled (hunk 2 above)
Comment (sponomarev) [reply bin/anycabled]: And this a most popular gem for daemonizing. Documentations is misleading. They handle ARGV for[ external scripts only](https://github.com/thuehlinger/daemons/blob/master/lib/daemons/application.rb#L223-L224). Such a shame. Anyway, it won't work [without bounded proc execution]( myproc.call).

Comment (sponomarev) [reply bin/anycabled]: Implemented in 43eb8a3

**Pull request merged**